packages/maskbook/src/extension/popups/pages/Wallet/ContractInteraction/index.tsx

```diff
@@ -0,0 +1,339 @@
+                    }
+                } else {
+                    // ERC20 Transfer and other contract interaction
+                    return {
+                        typeName: t('wallet_transfer_send'),
+                        spender: value.computedPayload._tx.from,
+                        to: value.computedPayload._tx.to,
+                        gasPrice: value.computedPayload._tx.gasPrice,
+                        maxFeePerGas: value.computedPayload._tx.maxFeePerGas,
+                        maxPriorityFeePerGas: value.computedPayload._tx.maxPriorityFeePerGas,
+                        amount: value.computedPayload.parameters?.value,
+                    }
+                }
+            case EthereumRpcType.SEND_ETHER:
+                return {
+                    typeName: t('wallet_transfer_send'),
+                    spender: value.computedPayload._tx.from,
+                    to: value.computedPayload._tx.to,
+                    gasPrice: value.computedPayload._tx.gasPrice,
+                    maxFeePerGas: value.computedPayload._tx.maxFeePerGas,
+                    maxPriorityFeePerGas: value.computedPayload._tx.maxPriorityFeePerGas,
+                    amount: value.computedPayload._tx.value,
+                }
+            default:
+                return {}
+        }
+    }, [value, t])
+
+    // const { value: token } = useERC20TokenDetailed(to)
+    const { value: nativeToken } = useNativeTokenDetailed()
+    //
+    // const { value: tokenPrices } = useAsync(async () => {
+    //     const coinList = await getAllCoins()
+    //
+    //     const tokenId = coinList?.find((coin) => coin.symbol === token?.symbol)?.id
+    //     const nativeTokenId = coinList?.find((coin) => coin.symbol === nativeToken?.symbol)?.id
+    //
+    //     if (!tokenId || !nativeTokenId)
+    //         return {
+    //             tokenPRice: 0,
+    //             nativeTokenPrice: 0,
+    //         }
+    //
+    //     const tokenPrice = await fetchTokenPrice(tokenId)
+    //     const nativeTokenPrice = await fetchTokenPrice(nativeTokenId)
+    //
+    //     return {
+    //         tokenPrice,
+    //         nativeTokenPrice,
+    //     }
+    // }, [token, nativeToken, value])
+    //
+    // const { tokenValueUSD, totalUSD } = useMemo(() => {
+    //     const tokenValueUSD = new BigNumber((amount ?? 0) as number).times(tokenPrices?.tokenPrice ?? 0).toString()
+    //
+    //     const totalUSD = formatWeiToEther(gasPrice as number)
+    //         .times(tokenPrices?.nativeTokenPrice ?? 0)
+    //         .plus(tokenValueUSD)
+    //
+    //     return {
+    //         tokenValueUSD,
+    //         totalUSD,
+    //     }
+    // }, [tokenPrices, gasPrice, amount])
+
+    const [{ loading }, handleConfirm] = useAsyncFn(async () => {
+        if (value) {
+            const toBeClose = new URLSearchParams(location.search).get('toBeClose')
+            await WalletRPC.deleteUnconfirmedRequest(value.payload)
```

Comment:
You need to reject the unconfirmed request if the confirmation dialog was closed by the user.